pkg/kubelet/kubelet_getters.go

```diff
@@ -281,12 +282,23 @@ func (kl *Kubelet) GetVersionInfo() (*cadvisorapiv1.VersionInfo, error) {
 
 // GetCachedMachineInfo assumes that the machine info can't change without a reboot
 func (kl *Kubelet) GetCachedMachineInfo() (*cadvisorapiv1.MachineInfo, error) {
-	if kl.machineInfo == nil {
+	return kl.machineInfo.getCachedMachineInfo(kl)
```

Comment:
do we still need this?

Reply:
talked offline. Remove this change, instead directly setting kl.machienInfo when newMainKubelet.